sites/fast-website/src/app/components/side-navigation/side-navigation.ts

```diff
@@ -14,25 +19,52 @@ export class SideNavigation extends FASTElement {
     @observable
     public sectionArray: HTMLElement[] = [];
 
+    @observable
+    public sectionDataArray: sectionData[];
+
+    @observable
+    public previousY: number = 0;
+
     public socialData: CommunityContentPlacementData[] = communityContentPlacementData.filter(
         x => x.header !== "Github"
     );
 
+    reduceFunc = (prev: sectionData, current: sectionData) =>
+        prev.intersectionRatio > current.intersectionRatio ? prev : current;
+
+    scrollDirectionFunc = () => {
+        const currentY: number = window.pageYOffset;
+        const scrollArray: sectionData[] = this.sectionDataArray;
```

Comment:
Just want to make sure you're intending to reverse `this.sectionDataArray` and not a just a local copy of the array

Reply:
I had been intending to just reverse the local copy. Thank you so much for explaining to me the current issues with that on discord. I have added .concat() to when I create the scrollArray variable so that it is a new array instance and does not update this.sectionDataArray.